baseapp/error.go

```diff
@@ -21,10 +21,17 @@ import (
 
 	"github.com/bluekeyes/hatpear"
 	"github.com/palantir/go-baseapp/pkg/errfmt"
+	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/hlog"
 )
 
+// HTTPError represents any error that presents itself as an HTTP error with a
+// status code.
+type HTTPError interface {
+	StatusCode() int
+}
```

Comment:
If we're going to call this `HTTPError`, it should probably embed `error`.

However, we could also follow the pattern used by `os.IsTimeout`, `errors.Cause`, and similar where the interface containing the `StatusCode` method is private and we document that `HandleError` will call `StatusCode` if it exists to get the code. I'm not sure how valuable it is to export this type if none of our APIs consume it directly.